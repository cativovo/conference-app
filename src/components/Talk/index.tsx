import React from 'react';

type TalkProps = {
  speakerName: string;
  name: string;
  description: string;
};

const Talk: React.FC<TalkProps> = ({ speakerName, name, description }) => (
  <div>
    <h3 data-testid="speaker-name">{speakerName}</h3>
    <h5 data-testid="name">{name}</h5>
    <p data-testid="description">{description}</p>
  </div>
);

export default Talk;
