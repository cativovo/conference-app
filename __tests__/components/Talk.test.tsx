import Talk from '../../src/components/Talk';
import { render } from '@testing-library/react';

it('should render correctly', () => {
  const props = {
    description: 'This is a test description',
    name: 'this is atest name',
    speakerName: 'This is a test speaker name',
  };
  const { container, queryByTestId } = render(<Talk {...props} />);

  const speakerName = queryByTestId('speaker-name');
  const name = queryByTestId('name');
  const description = queryByTestId('description');

  expect(container).toBeTruthy();
  expect(container.tagName).toBe('DIV');

  expect(speakerName).toBeTruthy();
  expect(speakerName?.tagName).toBe('H3');
  expect(speakerName?.textContent).toBe(props.speakerName);

  expect(name).toBeTruthy();
  expect(name?.tagName).toBe('H5');
  expect(name?.textContent).toBe(props.name);

  expect(description).toBeTruthy();
  expect(description?.tagName).toBe('P');
  expect(description?.textContent).toBe(props.description);

  expect(container).toMatchSnapshot();
});
