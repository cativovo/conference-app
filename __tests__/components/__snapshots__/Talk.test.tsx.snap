// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div>
    <h3
      data-testid="speaker-name"
    >
      This is a test speaker name
    </h3>
    <h5
      data-testid="name"
    >
      this is atest name
    </h5>
    <p
      data-testid="description"
    >
      This is a test description
    </p>
  </div>
</div>
`;
